src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncOptionsTest.java

```diff
@@ -0,0 +1,158 @@
+            mock(TaxCategorySyncOptionsTest.MockTriFunction.class);
+
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .beforeUpdateCallback(beforeUpdateCallback)
+            .build();
+
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = emptyList();
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertThat(filteredList).as("returned 'updateActions' should be empty").isEmpty();
+        verify(beforeUpdateCallback, never()).apply(any(), any(), any());
+    }
+
+    @Test
+    void applyBeforeUpdateCallback_WithCallback_ShouldReturnFilteredList() {
+        TriFunction<List<UpdateAction<TaxCategory>>, TaxCategoryDraft, TaxCategory, List<UpdateAction<TaxCategory>>>
+            beforeUpdateCallback = (updateActions, newCategory, oldCategory) -> emptyList();
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .beforeUpdateCallback(beforeUpdateCallback)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = singletonList(SetKey.of("key"));
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertAll(
+            () -> assertThat(filteredList)
+                .as("returned 'updateActions' should not be equal to prepared ones")
+                .isNotEqualTo(updateActions),
+            () -> assertThat(filteredList).as("returned 'updateActions' should be empty").isEmpty()
+        );
+    }
+
+    @Test
+    void applyBeforeCreateCallback_WithCallback_ShouldReturnFilteredDraft() {
+        Function<TaxCategoryDraft, TaxCategoryDraft> draftFunction =
+            taxCategoryDraft -> TaxCategoryDraftBuilder.of(taxCategoryDraft)
+                .key(taxCategoryDraft.getKey() + "_filteredKey").build();
+
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .beforeCreateCallback(draftFunction)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeCreateCallback()).isNotNull();
+
+        TaxCategoryDraft resourceDraft = mock(TaxCategoryDraft.class);
+        when(resourceDraft.getKey()).thenReturn("myKey");
+
+        Optional<TaxCategoryDraft> filteredDraft = taxCategorySyncOptions.applyBeforeCreateCallBack(resourceDraft);
+
+        assertAll(
+            () -> assertThat(filteredDraft).as("should return draft").isNotEmpty(),
+            () -> assertThat(filteredDraft.get().getKey())
```

Comment:
you can already use Optional assert:

````java
assertThat(filteredDraft)
            .hasValueSatisfying(draft -> assertThat(draft.getKey()).isEqualTo("myKey_filteredKey"));
````

Reply:
didn't know it is possible. thanks!